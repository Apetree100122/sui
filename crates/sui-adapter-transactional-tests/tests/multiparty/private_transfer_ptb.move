// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

//# init --addresses ex=0x0

//# publish
module ex::m;

public struct Priv has key {
    id: UID,
}

public fun mint(ctx: &mut TxContext) {
    let q = Priv { id: object::new(ctx) };
    transfer::transfer(q, ctx.sender());
}

public fun priv(ctx: &mut TxContext): Priv {
    Priv { id: object::new(ctx) }
}

//# run ex::m::mint

// Does not have store
//# programmable --inputs object(2,0) vector[@0]
//> sui::transfer::public_multiparty_transfer<ex::m::Priv>(Input(0), Input(1))

// Does not have store
//# programmable --inputs vector[@0]
//> 0: ex::m::priv();
//> sui::transfer::public_multiparty_transfer<ex::m::Priv>(Result(0), Input(0))

// Private transfer
//# programmable --inputs object(2,0) vector[@0]
//> sui::transfer::multiparty_transfer<ex::m::Priv>(Input(0), Input(1))

// Private transfer
//# programmable --inputs vector[@0]
//> 0: ex::m::priv();
//> sui::transfer::multiparty_transfer<ex::m::Priv>(Result(0), Input(0))
