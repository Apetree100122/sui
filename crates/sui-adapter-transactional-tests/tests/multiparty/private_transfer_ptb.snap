processed 7 tasks

task 1, lines 6-20:
//# publish
created: object(1,0)
mutated: object(0,0)
gas summary: computation_cost: 1000000, storage_cost: 5198400,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, lines 22-24:
//# run ex::m::mint
created: object(2,0)
mutated: object(0,0)
gas summary: computation_cost: 1000000, storage_cost: 2219200,  storage_rebate: 978120, non_refundable_storage_fee: 9880

task 3, lines 25-28:
//# programmable --inputs object(2,0) vector[@0]
//> sui::transfer::public_multiparty_transfer<ex::m::Priv>(Input(0), Input(1))
// Does not have store
Error: Transaction Effects Status: Move Bytecode Verification Error. Please run the Bytecode Verifier for more information.
Execution Error: ExecutionError: ExecutionError { inner: ExecutionErrorInner { kind: VMVerificationOrDeserializationError, source: Some(VMError { major_status: CONSTRAINT_NOT_SATISFIED, sub_status: None, message: None, exec_state: None, location: Module(ModuleId { address: sui, name: Identifier("transfer") }), indices: [], offsets: [] }), command: Some(0) } }

task 4, lines 29-33:
//# programmable --inputs vector[@0]
//> 0: ex::m::priv();
//> sui::transfer::public_multiparty_transfer<ex::m::Priv>(Result(0), Input(0))
// Private transfer
Error: Transaction Effects Status: Move Bytecode Verification Error. Please run the Bytecode Verifier for more information.
Execution Error: ExecutionError: ExecutionError { inner: ExecutionErrorInner { kind: VMVerificationOrDeserializationError, source: Some(VMError { major_status: CONSTRAINT_NOT_SATISFIED, sub_status: None, message: None, exec_state: None, location: Module(ModuleId { address: sui, name: Identifier("transfer") }), indices: [], offsets: [] }), command: Some(1) } }

task 5, lines 34-37:
//# programmable --inputs object(2,0) vector[@0]
//> sui::transfer::multiparty_transfer<ex::m::Priv>(Input(0), Input(1))
// Private transfer
Error: Transaction Effects Status: Non Entry Function Invoked. Move Call must start with an entry function
Execution Error: ExecutionError: ExecutionError { inner: ExecutionErrorInner { kind: NonEntryFunctionInvoked, source: Some("Cannot directly call sui::transfer::multiparty_transfer. Use the public variant instead, sui::transfer::public_multiparty_transfer"), command: Some(0) } }

task 6, lines 38-40:
//# programmable --inputs vector[@0]
//> 0: ex::m::priv();
//> sui::transfer::multiparty_transfer<ex::m::Priv>(Result(0), Input(0))
Error: Transaction Effects Status: Non Entry Function Invoked. Move Call must start with an entry function
Execution Error: ExecutionError: ExecutionError { inner: ExecutionErrorInner { kind: NonEntryFunctionInvoked, source: Some("Cannot directly call sui::transfer::multiparty_transfer. Use the public variant instead, sui::transfer::public_multiparty_transfer"), command: Some(1) } }
